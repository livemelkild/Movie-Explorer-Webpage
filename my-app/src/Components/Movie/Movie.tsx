import { Dispatch } from "redux";
import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { RootStore } from "../../store";
import { isPropertySignature } from "typescript";
import { selectModal } from "../../Reducer/ModalReducer";

import { openModal, fetchPending, fetchSuccess, fetchError } from '../../Action/Actions';

interface ModalState {
    id: number,
    title: string,
    year: string,
    user_rating: string,
    img_url: string,
    gendre: string[]
}

interface Rootreducer {

        searchReducer: {
            search: any;
        };
        loadingReducer: {
            loading: any;
        };
        fetchDataReducer: {
            fetch: any;
            pending: boolean;
            error: any;
        };
        sortReducer: {
            sortBy: string;
            order: number;
        };
        ModalReducer: ModalState;
}

interface Props {
    id: number,
    title: string,
    year: string,
    user_rating: string,
    img_url: string,
    gendre: string[]
}

export const Movie = () => {
    const dispatch = useDispatch();
    const searchState = useSelector((state: RootStore) => state.searchReducer.search);

    console.log(searchState);
    
    //kjører bare en gang, henter filmene til initalstate
    useEffect( 
        () => {
            console.log("search one time");
            dispatch(fetchPending());
            fetch('http://localhost:4000/api/movie')
            .then(response => response.json())
            .then((data: ModalState) => dispatch(fetchSuccess(data)))
            .catch(error => dispatch(fetchError))
             
              }, [] 
            )
    //kjører hver gang search state blir oppdatert
    useEffect(
        () => {
            console.log("try to search more times");
              }, [searchState]
            )
    
     //legger inn et bilde, endre til å hente bilde på ID         
    function parseUrl(id: number) {
        return `https://upload.wikimedia.org/wikipedia/commons/5/5b/Killbill-vol1-logo.svg`;
      }
    
    // henter selector i modelreducer
    const state = useSelector(selectModal);
    console.log(state.id);

    
    return ( 

        <div className="row">
        <div className="col s12 m7">
          <div className="card">
            <div className="card-image">
              <img src={parseUrl(state.id)} />
              <span className="card-title">Kill Bill</span>
            </div>
            <div className="card-content">
              <p>I am a very simple card. I am good at containing small bits of information.
              I am convenient because I require little markup to use effectively.</p>
            </div>
            <div className="card-action">
              <a><button onClick={() =>
                dispatch(
                    openModal({
                        id: state.id,
                        title: state.title,
                        year: state.year,
                        user_rating: state.user_rating,
                        img_url: state.img_url,
                        gendre: []
                    })
                )
            }> </button></a>
            </div>
          </div>
        </div>
      </div>
   )};

{/*
        <div className = "movieContainer"
            onClick={() =>
                dispatch(
                    openModal({
                        id: state.id,
                        title: state.title,
                        year: state.year,
                        user_rating: state.user_rating,
                        img_url: state.img_url,
                        gendre: []
                    })
                )
            }> 
            <div className='movieName'>Kill Bill</div> 
            <img className = "picture" src= {parseUrl(state.id)} />
        </div>
        */}

 
    

export default Movie;