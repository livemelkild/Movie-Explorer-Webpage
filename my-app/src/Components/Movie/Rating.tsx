import React, {useState} from "react";
import { setUpvote, removeUpvote } from "../../Action/Actions";
import { Dispatch } from "redux";
import { useDispatch, useSelector } from "react-redux";
import { RootStore } from "../../store";

import axios from 'axios'


interface iRating{
    title: string,
    upvote: number,
}


export default function Rating(props: iRating){
    const [title, setTitle] = useState(props.title)
    const dispatch: Dispatch<any> = useDispatch();

    const [ranking, setRanking] = useState(props.upvote);
    const [vote, setVote] = useState(0);

    const changeUpvote = (action: string )=> {
        if(action === "setUpvote"){
            if (vote == 1){
                console.log("already upvoted")
            }else{
                setVote(1)
                console.log(title)
                console.log("start opp" + ranking)

                axios.put(`http://localhost:4000/api/upVote/${title}`)

                const newRate = ranking + 1;
                
                setRanking(newRate);

                console.log("slutt opp" + ranking)

            }

        }else{
            if (vote == -1){
                console.log("already down voted")
            }else{
                setVote(-1)
                console.log("start ned" + ranking)
                axios.put(`http://localhost:4000/api/downVote/${title}`)
                
                const newRate = ranking -1;
                setRanking(newRate)
                console.log("slutt ned" + ranking)
            }


        }
    }



    return(
        
        <div>
            <button onClick = {() => changeUpvote("setUpvote")}>Upvote</button>
            <button onClick = {() => changeUpvote("removeUpvote")}>Down vote</button>
            
            <p>Movie rating: {ranking}</p>
            {}
        </div>

    )
    
}

