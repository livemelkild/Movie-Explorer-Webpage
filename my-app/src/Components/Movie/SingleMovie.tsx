import React from "react";
import "./SingleMovie.css"
import Rating from "./Rating"

interface iMovie {
        id: string | number,
        title: string | number,
        year: string | number,
        users_rating: string | number,
        img_url: string,
        genre: string
}

const SingleMovie = (props: iMovie) => {
const genreList = props.genre;

    return(
    <div className="container">
        <div className="col s12 m7">
          <div className="card">
            <div className="card-image">
              <img src= {props.img_url} alt="hei"/>
                <span className="card-title">{ props.title}</span>
            </div>
            <div className="card-content">
            <p> Year: {props.year}, Genre: {props.genre[0]}</p>
            <p>Antall upvote</p>
            </div>
            <div className="card-action">
              <Rating 
              id={props.id}/>
            </div>
          </div>
        </div>
    </div>
    )
}

export default SingleMovie;