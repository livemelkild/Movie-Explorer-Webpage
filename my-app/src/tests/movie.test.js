import React from 'react'
import renderer from 'react-test-renderer'


import Movie from "../components/Movie/Movie"

test('Movie snapshot test', () => {

    const tree = renderer.create(<Movie />)

    expect(tree).toMatchSnapshot()

});