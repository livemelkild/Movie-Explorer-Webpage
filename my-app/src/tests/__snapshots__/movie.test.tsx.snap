// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page snapshot test 1`] = `<test />`;
